lib/Sema/CSSimplify.cpp

```diff
@@ -6425,25 +6435,10 @@ static ConstraintFix *validateInitializerRef(ConstraintSystem &cs,
     // Initializer reference which requires contextual base type e.g.
     // `.init(...)`.
   } else if (auto *UME = getAsExpr<UnresolvedMemberExpr>(anchor)) {
-    // We need to find type variable which represents contextual base.
-    auto *baseLocator = cs.getConstraintLocator(
-        UME, locatorEndsWith(locator, ConstraintLocator::ConstructorMember)
-                 ? ConstraintLocator::UnresolvedMember
-                 : ConstraintLocator::MemberRefBase);
-
-    // FIXME: Type variables responsible for contextual base could be cached
-    // in the constraint system to speed up lookup.
-    auto result = llvm::find_if(
-        cs.getTypeVariables(), [&baseLocator](const TypeVariableType *typeVar) {
-          return typeVar->getImpl().getLocator() == baseLocator;
-        });
-
-    assert(result != cs.getTypeVariables().end());
-    baseType = cs.simplifyType(*result)->getRValueType();
+    auto baseType = cs.getUnresolvedMemberBaseType(UME);
```

Comment:
This isn't a perfect replacement for the logic to the left, but based on the comments it seems like it should work?